crates/router/src/core/recon.rs

```diff
@@ -138,29 +151,31 @@ pub async fn recon_merchant_account_update(
             format!("Failed while updating merchant's recon status: {merchant_id:?}")
         })?;
 
-    let user_email = &req.user_email.clone();
-    let email_contents = email_types::ReconActivation {
-        recipient_email: domain::UserEmail::from_pii_email(user_email.clone())
-            .change_context(errors::ApiErrorResponse::InternalServerError)
-            .attach_printable("Failed to convert recipient's email to UserEmail from pii::Email")?,
-        user_name: domain::UserName::new(Secret::new("HyperSwitch User".to_string()))
-            .change_context(errors::ApiErrorResponse::InternalServerError)
-            .attach_printable("Failed to form username")?,
-        settings: state.conf.clone(),
-        subject: consts::EMAIL_SUBJECT_APPROVAL_RECON_REQUEST,
-    };
-
-    if req.recon_status == enums::ReconStatus::Active {
-        let _ = state
-            .email_client
-            .compose_and_send_email(
-                Box::new(email_contents),
-                state.conf.proxy.https_url.as_ref(),
-            )
-            .await
-            .change_context(UserErrors::InternalServerError)
-            .attach_printable("Failed to compose and send email for ReconActivation")
-            .is_ok();
+    #[cfg(feature = "email")]
+    {
+        let user_email = &req.user_email.clone();
+        let email_contents = email_types::ReconActivation {
+            recipient_email: domain::UserEmail::from_pii_email(user_email.clone())
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable(
+                    "Failed to convert recipient's email to UserEmail from pii::Email",
+                )?,
+            user_name: domain::UserName::new(Secret::new("HyperSwitch User".to_string()))
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to form username")?,
+            subject: consts::EMAIL_SUBJECT_APPROVAL_RECON_REQUEST,
+        };
+        if req.recon_status == enums::ReconStatus::Active {
+            let _ = state
+                .email_client
+                .compose_and_send_email(
+                    Box::new(email_contents),
+                    state.conf.proxy.https_url.as_ref(),
+                )
+                .await
+                .change_context(UserErrors::InternalServerError)
+                .attach_printable("Failed to compose and send email for ReconActivation");
```

Comment:
By using the `let _` binding, we're ignoring the error occurring here, we're not even logging it. Is that intentional?

Reply:
Ignoring this error is intentional (it's for notifying the end user about recon activation for their account, which is not an API failure)

Reply:
We can choose to log the errors though